presto-iceberg/src/test/java/com/facebook/presto/iceberg/IcebergDistributedTestBase.java

```diff
@@ -2822,4 +2826,147 @@ protected void dropTable(Session session, String table)
         assertUpdate(session, "DROP TABLE " + table);
         assertFalse(getQueryRunner().tableExists(session, table));
     }
+
+    @Test
+    public void testEqualityDeleteAsJoinWithMaximumFieldsLimitUnderLimit()
+            throws Exception
+    {
+        int maxColumns = 10;
+        String tableName = "test_eq_delete_under_max_cols_" + randomTableSuffix();
+
+        Session session = Session.builder(getQueryRunner().getDefaultSession())
+                .setCatalogSessionProperty(ICEBERG_CATALOG, DELETE_AS_JOIN_REWRITE_ENABLED, "true")
+                // Make sure the max columns is set to one more than the number of columns in the table
+                .setCatalogSessionProperty(ICEBERG_CATALOG, DELETE_AS_JOIN_REWRITE_MAX_DELETE_COLUMNS, "" + maxColumns + 1)
```

Comment:
```suggestion
                .setCatalogSessionProperty(ICEBERG_CATALOG, DELETE_AS_JOIN_REWRITE_MAX_DELETE_COLUMNS, "" + (maxColumns + 1))
```

nit: Did you mean to set the upper limit to `maxColumns + 1`?

Reply:
Great catch!